File: 프로그래머스/3/43163. 단어 변환/단어 변환.cpp
#include <string>
#include <vector>

using namespace std;

bool canChange(string a, string b)
{
    int cnt = 0;
    for(int i = 0; i < a.size(); i++)
    {
        if(a[i]==b[i]) cnt++;
        if(cnt>1)return false;     
    }
    return true;
}

int solution(string begin, string target, vector<string> words) {
    
    int chanheCheck(){
        return 0
    }
    
    int answer = 0;
    return answer;
}

File: 프로그래머스/3/43163. 단어 변환/단어 변환.py
def canChange(a, b):
    count = 0
    for i in range(len(a)):
        if not a[i] == b[i]: count += 1
        if count > 1: return False
    return True  

def solution(begin, target, words):
    
    def changeCheck(begin, target, words, count): 
        #print(begin, count)
        
        # 변환 목표 도달
        if begin == target:
            return count
        
        minCnt = 1e9
        for w in words:
            if canChange(begin, w):
                tmp = words.copy()
                tmp.remove(w)
                res = changeCheck(w, target, tmp, count + 1)
                if not res == None:
                    minCnt = min(minCnt, res)
                    #print(minCnt)
                            
        return minCnt
    
    cnt = changeCheck(begin, target, words, 0)
    if cnt == 1e9: return 0
    return cnt